packages/odyssey-react/src/components/Text/Text.tsx

```diff
@@ -97,7 +104,42 @@ export interface TextProps {
    * The overflow wrapping behavior for the text content.
    * @default normal
    */
-  wrap?: "normal" | "breakWord" | "anywhere";
+  wrap?: "normal" | "breakWord" | "anywhere" | "nowrap";
+
+  /**
+   * The text color style on hover.
+   */
+  hoverColor?: "body" | "bodyInverse" | "primaryDark";
+
+  /**
+   * The text color style on focus.
+   */
+  focusColor?: "body" | "bodyInverse" | "primary";
+
+  /**
+   * The text color style when disabled.
+   */
+  disabledColor?: "bodyInverse" | "primaryLight" | "sub";
+
+  /**
+   * The text color style when the parent element is hovered.
+   */
+  parentHoveredColor?: "body" | "bodyInverse" | "primaryDark";
+
+  /**
+   * The text color style when the parent element has focus.
+   */
+  parentFocusedColor?: "body" | "bodyInverse" | "primary";
+
+  /**
+   * The text color style when the parent element is disabled.
+   */
+  parentDisabledColor?: "bodyInverse" | "primaryLight" | "sub";
```

Comment:
Have we discussed other ways to solve this as a group yet? This API design does not feel right and I'd love to get on the same page quickly this week. If the parent is disabled or focused can it then be in charge of applying the corect variant into Text rather than relying on unscoped psuedo-attribue descendant selector styles and a `parent*` prop based API like this?

Reply:
This speaks to my usage questions in Slack and here: https://github.com/okta/odyssey/pull/1176#discussion_r749574434

If `Text` is intended for inserting and styling semantic elements, these concerns can be left to the parent, as the styles are applied to a leaf. In that case, Button does not require Text.

If `Text` is intended to be a leaf-styling utility, these concerns should be addressed by `Text`, so that it's a complete utility with use. In that case, `Text` should not need to know about any parent component concerns like `background-color`.

If `Text` is intended to be an inheritance source (as in `Infobox`), the other two cases become obsolete but behaviors cannot be component-specific, making Text untenable for Button.

Currently, we use `Text` for all of these cases, resulting in a confused API design.

Reply:
@edburyenegren-okta what solution are proposing here?

Reply:
Proposal based on this morning's convo: https://c198ad7.ods.dev/?path=/story/utilities-utilities--page#box